esmvalcore/_recipe/check.py

```diff
@@ -391,20 +400,24 @@ def _check_format_years(date):
 
 
 def _check_timerange_values(date, timerange):
+    # P must always be in a duration string
+    # if T in date, that is a datetime; otherwise it's date
```

Comment:
```suggestion
    # Wildcards are fine
    if date == '*':
        return
    
    # P must always be in a duration string
    # if T in date, that is a datetime; otherwise it's date
```

feels more natural